### PR TITLE
SAK-48233 - Lessons: Add the maximum allowed file size when embedding…

### PR DESCRIPTION
**Pull request by JuanDavid102**
… files

### CODE REVIEW

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java [hunk 1]
```diff
@@ -4616,6 +4616,20 @@ private void createAnnouncementsDialog(UIContainer tofill, SimplePage currentPag
 	// in the edit dialogs
 	private void createAddMultimediaDialog(UIContainer tofill, SimplePage currentPage) {
 		UIOutput.make(tofill, "add-multimedia-dialog").decorate(new UIFreeAttributeDecorator("title", messageLocator.getMessage("simplepage.resource")));
+
+		String max = ServerConfigurationService.getString("content.upload.max", null);
```
Comment (jesusmmp) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java:4620]: String max = ServerConfigurationService.getString("content.upload.max", 20);

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java [hunk 2]
```diff
@@ -4616,6 +4616,20 @@ private void createAnnouncementsDialog(UIContainer tofill, SimplePage currentPag
...
+		String ceiling = ServerConfigurationService.getString("content.upload.ceiling", null);
```
Comment (jesusmmp) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java:4621]: String ceiling = ServerConfigurationService.getString("content.upload.ceiling", 20);

**Comment by jesusmmp** [comment]
Better to take default values which are defined on sakai.properties

# Upload limit per request, in MBs.
# "ceiling" is used for resources. "max" is used for attachments. 
# In the case when both are set, resources are limited to the smaller of the two. If only one of the two are set, resource uploads
# are limited to the value whichever is set. Attachments are limited at either 20 or the setting of "max".
# Example: content.upload.max=50 and content.upload.ceiling=25 (attachments limit of 50MB and resource uploads at 25MB)
# Example: content.upload.max=10 and content.upload.ceiling=15 (attachments limit of 10MB and resource uploads at 10MB)
# Example: content.upload.ceiling=25 (attachments limit of 20MB and resource uploads at 25MB)
# Example: content.upload.max=50 (attachments limit of 50MB and resource uploads at 50MB)
# DEFAULT: 20
# content.upload.max=20
# **Note - online documentation states ceiling is set in bytes - this is incorrect. Settings are in MB.
# content.upload.ceiling=20
# The UI should show hidden as the default when uploading files
# content.dnd.visibility.hidden=false

**Comment by mpellicer** [comment]
@JuanDavid102 I made similar comments in the private PR, take a look.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java (hunk 1 above)
Comment (mpellicer) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java:4620]: The it should be:

int max = ServerConfigurationService.getInt("content.upload.max", 20);

Or 

String max = ServerConfigurationService.getString("content.upload.max", "20");

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java [hunk 3]
```diff
@@ -4616,6 +4616,13 @@ private void createAnnouncementsDialog(UIContainer tofill, SimplePage currentPag
 	// in the edit dialogs
 	private void createAddMultimediaDialog(UIContainer tofill, SimplePage currentPage) {
 		UIOutput.make(tofill, "add-multimedia-dialog").decorate(new UIFreeAttributeDecorator("title", messageLocator.getMessage("simplepage.resource")));
+
+		String max = ServerConfigurationService.getString("content.upload.max", 20);
+		String ceiling = ServerConfigurationService.getString("content.upload.ceiling", max);
+		String uploadMax = ceiling;
```
Comment (mpellicer) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java:4622]: tabs vs spaces.

Comment (mpellicer) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java:4622]: This does not compile, 

String max = ServerConfigurationService.getString("content.upload.max", "20");

**Review by mpellicer — CHANGES_REQUESTED** [review]
I don't think this is working, I don't see any change in the markup.

**Review by mpellicer — APPROVED** [review]

**Comment by mpellicer** [comment]
This works fine thanks @JuanDavid102 !

image

**Review by jesusmmp — APPROVED** [review]

**Pull request merged**